### PR TITLE
[#10919] Deprecate Map.{filterKeys,mapValues} while undeprecating those in MapView

### PR DESCRIPTION
**Pull request by joshlemer**
This implements scala/bug#10919

### CODE REVIEW

**Comment by lrytz** [comment]
A number of tests now issue deprecation warnings (https://scala-ci.typesafe.com/job/scala-2.13.x-validate-main/3621/), they need to be updated to use `.view.X`

File: src/library/scala/collection/Map.scala [hunk 1]
```diff
@@ -193,13 +193,15 @@ trait MapOps[K, +V, +CC[_, _] <: IterableOps[_, AnyConstr, _], +C]
     *  @return an immutable map consisting only of those key value pairs of this map where the key satisfies
     *          the predicate `p`. The resulting map wraps the original map without copying any elements.
     */
+  @deprecated("Use view.filterKeys. A future version will include a strict version of this method.", "2.13.0")
```
Comment (julienrf) [inline src/library/scala/collection/Map.scala]: I would say:

> Use `.view.filterKeys(p)`.

(The second sentence looks good)

We could maybe also document how to perform this operation in a strict way (`.view.filterKeys(p).toMap`)?

Comment (joshlemer) [reply src/library/scala/collection/Map.scala]: @julienrf I'm on it

**Comment by joshlemer** [comment]
@julienrf this is ready for review

File: src/library/scala/collection/concurrent/TrieMap.scala [hunk 2]
```diff
@@ -951,14 +951,14 @@ final class TrieMap[K, V] private (r: AnyRef, rtupd: AtomicReferenceFieldUpdater
...
-  override def filterKeys(p: K => Boolean): collection.MapView[K, V] = {
-    if (nonReadOnly) readOnlySnapshot().filterKeys(p)
-    else super.filterKeys(p)
-  }
-  override def mapValues[W](f: V => W): collection.MapView[K, W] = {
-    if (nonReadOnly) readOnlySnapshot().mapValues(f)
-    else super.mapValues(f)
-  }
+
+  override def view: MapView[K, V] = if (nonReadOnly) readOnlySnapshot().view else super.view
```
Comment (julienrf) [inline src/library/scala/collection/concurrent/TrieMap.scala:955]: Good catch!

**Review by julienrf — APPROVED** [review]
Awesome @joshlemer, thanks!

File: src/library/scala/collection/Map.scala [hunk 3]
```diff
@@ -193,13 +193,15 @@ trait MapOps[K, +V, +CC[_, _] <: IterableOps[_, AnyConstr, _], +C]
...
+  @deprecated("Use .view.filterKeys(f). A future version will include a strict version of this method (for now, .view.filterKeys(p).toMap).", "2.13.0")
   def filterKeys(p: K => Boolean): MapView[K, V] = new MapView.FilterKeys(this, p)
 
   /** Transforms this map by applying a function to every retrieved value.
     *  @param  f   the function used to transform values of this map.
     *  @return a map view which maps every key of this map
     *          to `f(this(key))`. The resulting map wraps the original map without copying any elements.
     */
+  @deprecated("Use .view.mapValues(f). A future version will include a strict version of this method (for now, .view.mapValues(f).toMap).", "2.13.0")
   def mapValues[W](f: V => W): MapView[K, W] = new MapView.MapValues(this, f)
```
Comment (joshlemer) [inline src/library/scala/collection/Map.scala:205]: Maybe I should change the implementations of these to `.view.mapValues(f)` and `.view.filterKeys(p)`

Comment (julienrf) [reply src/library/scala/collection/Map.scala:205]: I think that’s fine as it is.

File: src/library/scala/collection/concurrent/TrieMap.scala (hunk 2 above)
Comment (joshlemer) [reply src/library/scala/collection/concurrent/TrieMap.scala:955]: All thanks too the existing exhaustive tests :-)

**Comment by SethTisue** [comment]
thank you Josh!

**Pull request merged**